passes/source2il.nim

```diff
@@ -167,6 +221,22 @@ proc exprToIL(c; t: InTree, n: NodeIndex, bu): TypeKind =
       result = tkError
   of SourceKind.Call:
     result = callToIL(c, t, n, bu)
+  of SourceKind.Return:
+    var typ: TypeKind
+    bu.subTree Return:
+      typ =
+        case t.len(n)
+        of 0: tkUnit
+        of 1: exprToIL(c, t, t.child(n, 0), bu)
+        else: unreachable() # syntax error
+
+    if typ notin {tkError, tkVoid} and typ == c.retType:
+      result = tkVoid
+    else:
+      # TODO: use proper error correction; a return expression is always of
+      #       type ``void``
```

Comment:
This requires a proper error reporting facility first. At the moment, returning `tkError` is the only way for `exprToIL` to signal that an error occurred somewhere.